graph/articles/collections.md

```diff
@@ -316,3 +316,151 @@ Content-Type: application/json
+
+```xml
+<EntityType Name="application">
+  <Key>
+    <PropertyRef Name="id" />
+  </Key>
+  <Property Name="id" Type="Edm.String" Nullable="false" />
+  <Property Name="keyCredentials" Type="Collection(self.keyCredential)" />
+  ...
+</EntityType>
+
+<ComplexType Name="keyCredential">
+  <Property Name="keyId" Type="Edm.Guid" />
+  <Property Name="endDateTime" Type="Edm.DateTimeOffset" />
+  ...
+</ComplexType>
+```
+and a scenario arises that requires, for example, to remove individual `keyCredential`s from the collection. 
+There are two options forward: //// TODO do we want to offer both options, or just one?
+
+### 11.1 Side-by-side collection properties
+
+The model can be updated to have two collections side-by-side:
+```diff
+<EntityType Name="application">
+  <Key>
+    <PropertyRef Name="id" />
+  </Key>
+  <Property Name="id" Type="Edm.String" Nullable="false" />
+  <Property Name="keyCredentials" Type="Collection(self.keyCredential)" />
++ <NavigationProperty Name="keyCredentialsAsEntities" Type="Collection(self.keyCredentialAsEntity)" ContainsTarget="true" />
+</EntityType>
+
+<ComplexType Name="keyCredential">
+  <Property Name="keyId" Type="Edm.Guid" />
+  <Property Name="endDateTime" Type="Edm.DateTimeOffset" />
+</ComplexType>
+
++<EntityType Name="keyCredentialAsEntity">
++ <Key>
++   <PropertyRef Name="keyId" />
++ </Key>
++ <Property Name="keyId" Type="Edm.Guid" />
++ <Property Name="endDateTime" Type="Edm.DateTimeOffset" />
++</EntityType>
+```
+Clients will now be able to refer to individual `keyCredential`s using `keyId` as a key, and they can now remove those `keyCredential`s using `DELETE` requests:
+```http
+DELETE /applications/{applicationId}/keyCredentialsAsEntities/{some_keyId}
+```
+```http
+HTTP/1.1 204 No Content
+```
+The expectation is that `keyCredentials` and `keyCredentialsAsEntities` are treated as two "views" into the same data.
+To meet this expectation, workloads must:
+1. Keep the properties consistent between `keyCredential` and `keyCredentialAsEntity`.
+Any changes to one type must be reflected in the other type.
+2. Reject requests that update both collections at the same time.
+A request that adds an item to `keyCredentialsAsEntities` while replacing the content of `keyCredentials` must rejected with a `400`, for example:
+```http
+PATCH /applications/{applicationId}
+{
+  "keyCredentials": [
+    {
+      "keyId": "10000000-0000-0000-0000-000000000000",
+      "endDateTime": "2012-12-03T07:16:23Z"
+    }
+  ],
+  "keyCredentialsAsEntities@delta": [
+    {
+      "keyId": "20000000-0000-0000-0000-000000000000",
+      "endDateTime": "2012-12-03T07:16:23Z"
+    }
+  ]
+}
+```
+```http
+HTTP/1.1 400 Bad Request
+{
+  "error": {
+    "code": "badRequest",
+    "message": "'keyCredentials' and 'keyCredentialsAsEntities' cannot be updated in the same request.",
+}
+```
+
+TODO should this be a 409 conflict instead?
+TODO implement this in WebApi
+
+### 11.2 `$select` overloading
```

Comment:
I'm not sure that calling it "$select overloading" conveys the intent of this option -- i.e., if I was thinking about different ways to evolve the collection, I wouldn't look for an option called "Select overloading". That is more of a by-product than an option. The option is to replace the complex type with an entity type -- in order for this to work, however, existing select semantics need to be maintained.

```suggestion
### 11.2 Redefine as Entity Type
```